feat(arbitrum): add arbitrum_one

alfetopito opened the pull request:
# Summary

Part of #4310

Still a work in progress

# To Test

Full regression testing on the whole app on all networks 😬

alfetopito commented on libs/tokens/src/services/fetchTokenList.ts:
```diff
@@ -63,13 +70,32 @@ async function _fetchTokenList(source: string, urls: string[]): Promise<ListStat
+        source,
+        list: await sanitizeList(json),
+      }
+    } catch (e) {
+      const message = `failed to process list ${url}`
+      console.debug(message, e)
+
+      if (isLast) throw new Error(message)
+
+      continue
```
This check didn't exist. If a list was invalid, it would break the loop and not continue with the remaining.

alfetopito commented on apps/cowswap-frontend/src/modules/permit/const.ts:
```diff
@@ -2,5 +2,6 @@ import ms from 'ms.macro'
 export const PENDING_ORDER_PERMIT_CHECK_INTERVAL = ms`60s`
 
+// TODO: revert to previous one, only for temporary testing
 export const PRE_GENERATED_PERMIT_URL =
-  'https://raw.githubusercontent.com/cowprotocol/token-lists/main/src/public/PermitInfo'
+  'https://raw.githubusercontent.com/cowprotocol/token-lists/2a83e3299264b31b99e3bc760223cb75b14fe7fa/src/public/PermitInfo'
```
Should revert once https://github.com/cowprotocol/token-lists/pull/389 is merged

elena-zh commented:
Hey @alfetopito ,great job!
Here I will list issues I've faced (n addition to the ones reported in https://cowservices.slack.com/archives/C0361CDG8GP/p1715876405779169
1. When I open Favourites tokens in Account page. I see loaders for USDT token 
![image](https://github.com/cowprotocol/cowswap/assets/70885163/276d2c83-7b4c-4bd1-a75a-1b7f288132f7)
2. (not sure), but for the Surplus in Arb it would be great to change the tooltip saying from May 2024 :) 
![image](https://github.com/cowprotocol/cowswap/assets/70885163/1787e4b7-95a8-48f9-86de-520ddbbb9208)
3. Should we change 'View on Etherscan' to 'View on Arbiscan' everywhere in the app?
![image](https://github.com/cowprotocol/cowswap/assets/70885163/3b9c82c1-643d-4aa5-ab87-a2179d885e63)
4. ETH-flow is not working 
![image](https://github.com/cowprotocol/cowswap/assets/70885163/a722063d-3c16-4b40-a871-d6bef97a1b56)
5. Classic ETH-flow is also not proposed when fill in ETH token into the sell field
![image](https://github.com/cowprotocol/cowswap/assets/70885163/7fee4dbe-a8b2-49c9-a5f9-ccd4a30c5985)
6. Same with Safe +WC: Swap button remains disabled instead of showing Wrap+Swap
7. Update tokens with the new network on cow.fi/tokens page? https://cow.fi/tokens/cow-protocol
![image](https://github.com/cowprotocol/cowswap/assets/70885163/68bfbbba-87c6-43c2-9869-054cc8114d1b)
![image](https://github.com/cowprotocol/cowswap/assets/70885163/00b571f0-16a3-4eb7-b292-bdca1061fbef)

8. Update widget configurator 
![image](https://github.com/cowprotocol/cowswap/assets/70885163/a8f48672-6cc8-42ad-9c45-e240e0b95ac7)
9. Update docs
10. Partner fee?

Will continue tests in other networks tomorrow and will wait for all the rest fixes.

alfetopito commented:
@elena-zh 

1. fixed
2. fixed
3. fixed
4. fixed
5. fixed
6. didn't check but should fixed with the ethflow fix
7. not fixed yet, will leave this for later
8. fixed
9. not fixed yet, will leave this for later
10. should accept arbitrum already

shoom3301 commented on libs/wallet/src/web3-react/utils/switchChain.ts:
```diff
@@ -11,6 +11,8 @@ import { getIsWalletConnect } from '../hooks/useIsWalletConnect'
 function getRpcUrls(chainId: SupportedChainId): [string] {
   if (chainId === SupportedChainId.GNOSIS_CHAIN) {
     return ['https://rpc.gnosischain.com/']
+  } else if (chainId === SupportedChainId.ARBITRUM_ONE) {
```
Let's make `WALLET_SUGGESTION_RPCS` const with `Record<SupportedChainId, string | null>` type?

shoom3301 commented on libs/common-utils/src/legacyAddressUtils.ts:
```diff
@@ -114,6 +114,8 @@ export function getExplorerLabel(chainId: SupportedChainId, type: BlockExplorerL
     return 'View on Explorer'
   } else if (chainId === SupportedChainId.GNOSIS_CHAIN) {
     return 'View on Gnosisscan'
+  } else if (chainId === SupportedChainId.ARBITRUM_ONE) {
```
```
if (isCowOrder(type, data)) {
    return 'View on Explorer'
}

return `View on ${CHAIN_INFO[chainId].explorerTitle}`
```

shoom3301 commented on libs/common-utils/src/explorer.ts:
```diff
@@ -50,6 +51,7 @@ enum Explorers {
+  Arbiscan = 'Arbiscan',
 }
 
 // Used for GA ExternalLink detection
```
The function is only used in `externalLinkAnalytics`, I'm curious, do we still need the event?

alfetopito replied:
I think so?

![image](https://github.com/cowprotocol/cowswap/assets/43217/153d781a-a651-419d-8267-e791fd0719fd)

Not 100% sure

elena-zh commented:
Hey @alfetopito , great, great job!

For now, I've faced these issues during testing: 
1. When I switch to Arb in MM,  right after the switching it asks me to return back to the previously selected network in the wallet: 
![image](https://github.com/cowprotocol/cowswap/assets/70885163/2faf14fb-e808-426c-b81a-1a8cb752538a)
It would be great to stay connected to Arb after adding it to the wallet. 
2. Is this OK that I see error in console after signing ETH order (or any other onchain TX)?
![image](https://github.com/cowprotocol/cowswap/assets/70885163/1b760bdc-1b97-4c1c-a88a-dc469dd7a6ad)
3. (later): update Terms? Some examples: 
![image](https://github.com/cowprotocol/cowswap/assets/70885163/ba2916ad-6541-4799-b737-8c6df0d4c12b)
![image](https://github.com/cowprotocol/cowswap/assets/70885163/20efb97b-b512-4462-a1a6-559758150ae9)
4. Issue reported by you: https://cowservices.slack.com/archives/C0361CDG8GP/p1716220143272649 :) 


Tomorrow will proceed with tests in other networks.
Thanks!

anxolin commented on apps/cowswap-frontend/src/legacy/components/CowSubsidyModal/SubsidyTable.tsx:
```diff
@@ -163,6 +162,8 @@ const vCowToken = V_COW[SupportedChainId.MAINNET]
 function SubsidyTable({ discount }: CowSubsidy) {
   const darkMode = useIsDarkMode()
 
+  if (!vCowToken) return null
```
cool, glad we can't opt out of this

anxolin commented on apps/cowswap-frontend/src/pages/Account/LockedGnoVesting/claimData/index.ts:
```diff
@@ -12,12 +12,14 @@ interface Claim {
 const chainNames: Record<SupportedChainId, string> = {
   [SupportedChainId.MAINNET]: 'mainnet',
   [SupportedChainId.GNOSIS_CHAIN]: 'gnosisChain',
+  [SupportedChainId.ARBITRUM_ONE]: '',
```
can you use undefined instead?

anxolin commented on apps/explorer/src/components/common/BlockExplorerLink/BlockExplorerLink.tsx:
```diff
@@ -52,15 +53,15 @@ export const BlockExplorerLink: React.FC<Props> = (props: Props) => {
       {showLogo && (
         <LogoWrapper
-          title={`Open it on ${networkId === Network.GNOSIS_CHAIN ? 'Gnosisscan' : 'Etherscan'}`}
+          title={`Open it on ${CHAIN_INFO[networkId].explorerTitle}`}
```
ubernit: Suince there's these url and label const there, I would think is nicer to restructure the `CHAIN_INFO[networkId]` so the markup is simpler

anxolin commented on apps/explorer/src/explorer/pages/Home/index.tsx:
```diff
@@ -55,7 +55,8 @@ const SummaryWrapper = styled.section`
   [SupportedChainId.MAINNET]: true,
   [SupportedChainId.GNOSIS_CHAIN]: false, // Gchain data is not reliable
-  [SupportedChainId.SEPOLIA]: false, // No data for Sepolia
+  [SupportedChainId.ARBITRUM_ONE]: false,
```
should  we add a comment since we do it for GC? Like: `The graph API is not available for this network`

anxolin commented on libs/common-const/src/common.ts:
```diff
@@ -52,32 +53,27 @@ export const APP_TITLE = 'CoW Swap | The smartest way to trade cryptocurrencies'
 export const V_COW_CONTRACT_ADDRESS: Record<SupportedChainId, string> = {
   [SupportedChainId.MAINNET]: '0xd057b63f5e69cf1b929b356b579cba08d7688048',
   [SupportedChainId.GNOSIS_CHAIN]: '0xc20C9C13E853fc64d054b73fF21d3636B2d97eaB',
+  [SupportedChainId.ARBITRUM_ONE]: '', // doesn't exist!
```
Can you use `undefined` or `null` here?

anxolin commented on libs/common-const/src/common.ts:
```diff
@@ -52,32 +53,27 @@ export const APP_TITLE = 'CoW Swap | The smartest way to trade cryptocurrencies'
 export const COW_CONTRACT_ADDRESS: Record<SupportedChainId, string> = {
   [SupportedChainId.MAINNET]: '0xDEf1CA1fb7FBcDC777520aa7f396b4E015F497aB',
   [SupportedChainId.GNOSIS_CHAIN]: '0x177127622c4A00F3d409B75571e12cB3c8973d3c',
+  [SupportedChainId.ARBITRUM_ONE]: '0xcb8b5cd20bdcaea9a010ac1f8d835824f5c87a04',
```
I was surprised to see the name of our precious COW contract: `ClonableBeaconProxy`

I guess this is the implementation of all the bridged tokens :)

anxolin reviewed (commented):
Soft approve! with some minor comments

elena-zh commented:
Hey @alfetopito , great job!
All issues I've reported above are fixed. 
I'm waiting for [other ](https://github.com/cowprotocol/cowswap/pull/4407#issue-2278081403) known issues to be fixed, then will approve the current PR :)

shoom3301 commented:
@alfetopito not part of this PR, but along with releasing it we should update docs: https://docs.cow.fi/governance/token

elena-zh commented:
Great job!
